test/client-unit/services/PurchaseOrderFormService.spec.js

```diff
@@ -0,0 +1,148 @@
+/* global inject, expect, chai */
+/* eslint no-unused-expressions:off, no-restricted-properties:off */
+describe('PurchaseOrderForm', () => {
+  let PurchaseOrderForm;
+  let Mocks;
+  let Session;
+  let httpBackend;
+  let $timeout;
+  let form;
+
+  beforeEach(module(
+    'bhima.services',
+    'angularMoment',
+    'ui.bootstrap',
+    'ui.router',
+    'ngStorage',
+    'pascalprecht.translate',
+    'tmh.dynamicLocale',
+    'bhima.StockMocks',
+    'bhima.mocks',
```

Comment:
Since you asked in your comments - nope, this is the minimal set of all possible modules.  I tried to tweak it manually, but you need each one of these.